clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -453,6 +476,42 @@ static CXXMethodDecl *getInitMethod(const CXXRecordDecl *CRD) {
   return InitMethod;
 }
 
+static CXXMethodDecl *getOperatorParenthesis(CXXRecordDecl *FuncObjType) {
+  const auto Methods = FuncObjType->methods();
+  const auto PredF = [=](const CXXMethodDecl *MI) -> bool {
+    return (MI->getOverloadedOperator() == OO_Call);
+  };
+  const auto Res = std::find_if(Methods.begin(), Methods.end(), PredF);
+  assert(((Res == Methods.end()) ||
```

Comment:
Where do we make sure this doesn't happen? multiple operator calls are entirely possible in user defined types.

Reply:
This question triggered some rethinking on my side :) You are right, single '()' operator is expected only if we have lambda form of the kernel. With functors many of those are possible.(with only one being suitable for current kernel invocation).  I reworked the way the proper '()' is found and updated the unit test.